pkg/prometheus/statefulset.go

```diff
@@ -1076,3 +676,488 @@ func queryLogFilePath(queryLogFile string) string {
+func buildCommonVolumes(cpf monitoringv1.CommonPrometheusFields, promName, volName string, tlsAssetSecrets []string) ([]v1.Volume, []v1.VolumeMount, error) {
+	assetsVolume := v1.Volume{
+		Name: "tls-assets",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{},
+			},
+		},
+	}
+	for _, assetShard := range tlsAssetSecrets {
+		assetsVolume.Projected.Sources = append(assetsVolume.Projected.Sources,
+			v1.VolumeProjection{
+				Secret: &v1.SecretProjection{
+					LocalObjectReference: v1.LocalObjectReference{Name: assetShard},
+				},
+			})
+	}
+
+	volumes := []v1.Volume{
+		{
+			Name: "config",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: configSecretName(promName),
+				},
+			},
+		},
+		assetsVolume,
+		{
+			Name: "config-out",
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{
+					// tmpfs is used here to avoid writing sensitive data into disk.
+					Medium: v1.StorageMediumMemory,
+				},
+			},
+		},
+	}
+
+	if cpf.Storage != nil {
+		if cpf.Storage.VolumeClaimTemplate.Name != "" {
+			volName = cpf.Storage.VolumeClaimTemplate.Name
+		}
+	}
+
+	promVolumeMounts := []v1.VolumeMount{
+		{
+			Name:      "config-out",
+			ReadOnly:  true,
+			MountPath: confOutDir,
+		},
+		{
+			Name:      "tls-assets",
+			ReadOnly:  true,
+			MountPath: tlsAssetsDir,
+		},
+		{
+			Name:      volName,
+			MountPath: storageDir,
+			SubPath:   subPathForStorage(cpf.Storage),
+		},
+	}
+
+	promVolumeMounts = append(promVolumeMounts, cpf.VolumeMounts...)
+
+	// Mount related secrets
+	rn := k8sutil.NewResourceNamerWithPrefix("secret")
+	for _, s := range cpf.Secrets {
+		name, err := rn.DNS1123Label(s)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		volumes = append(volumes, v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: s,
+				},
+			},
+		})
+		promVolumeMounts = append(promVolumeMounts, v1.VolumeMount{
+			Name:      name,
+			ReadOnly:  true,
+			MountPath: secretsDir + s,
+		})
+	}
+
+	rn = k8sutil.NewResourceNamerWithPrefix("configmap")
+	for _, c := range cpf.ConfigMaps {
+		name, err := rn.DNS1123Label(c)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		volumes = append(volumes, v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: c,
+					},
+				},
+			},
+		})
+		promVolumeMounts = append(promVolumeMounts, v1.VolumeMount{
+			Name:      name,
+			ReadOnly:  true,
+			MountPath: configmapsDir + c,
+		})
+	}
+
+	return volumes, promVolumeMounts, nil
+}
+
+// addServerVolumes adds volumes to the statefulset spec that are specific to Prometheus Server
+func addServerVolumes(volumes *[]v1.Volume, volumeMounts *[]v1.VolumeMount, queryLogFile string, ruleConfigMapNames []string) {
```

Comment:
same suggestion about the function signature
```suggestion
func appendServerVolumes(volumes []v1.Volume, volumeMounts *[]v1.VolumeMount, queryLogFile string, ruleConfigMapNames []string) []v1.Volume {
```